slice/validate.go

```diff
@@ -2,17 +2,12 @@ package slice
 
 import (
 	"fmt"
+	"regexp"
 )
 
-func (s *Split) init() error {
-	if len(s.opts.IncludedKinds) > 0 && len(s.opts.ExcludedKinds) > 0 {
-		return fmt.Errorf("cannot specify both included and excluded kinds")
-	}
-
-	if len(s.opts.IncludedNames) > 0 && len(s.opts.ExcludedNames) > 0 {
-		return fmt.Errorf("cannot specify both included and excluded names")
-	}
+const regKN = `^[^/]+/[^/]+$`
```

Comment:
Since this string will never change, below during the call of `regexp.MatchString()` the regexp will be compiled every time that function is called. Granted, it's a CLI and not a long-running app, but we can still optimize those.

May I propose a different alternative?

* Compile the regexp at the package level; then
* Use the regexp

A conthrived example:

```golang
package xyz

// declare compiled regexp here
var reKN = regexp.MustCompile(`^[^/]+/[^/]+$`)

func doesItMatch(str string) bool {
  // use it here without having to recompile it
  return reKN.MatchString(str)
}
```

It will also save some indentation lines and if we ever get to make a mistake in the regexp and it doesn't compile, we'll quickly know during testing.

Reply:
It seems the CI linter also found the same issue:

```
  SA6000: calling regexp.MatchString in a loop has poor performance, consider using regexp.Compile (staticcheck)
```

Imho, it's better in this case to use `regexp.MustCompile()` instead and then just called the precompiled regexp.

Reply:
I totally agree, to be honest, I didn't use the `regexp` package quite often, so I didn't have any idea about this. Thanks <3

Reply:
Done!